pytket/qir/conversion/conversion.py

```diff
@@ -244,6 +251,42 @@ def __init__(
             self.circuit.n_qubits + 1
         )
 
+        # void functionname()
+        if wfh is not None:
+            self.wasm: dict[str, pyqir.Function] = {}
+            for fn in wfh._functions:
+                wasm_func_interface = "declare "
+                parametertype = [self.qir_int_type] * wfh._functions[fn][0]
+                if wfh._functions[fn][1] == 0:
+                    returntype = pyqir.Type.void(self.module.module.context)
+                    wasm_func_interface += "void "
+                elif wfh._functions[fn][1] == 1:
+                    returntype = self.qir_int_type
+                    wasm_func_interface += f"i{self.int_type_str} "
+                else:
+                    raise ValueError(
+                        "wasm function which return more than"
```

Comment:
Yes I will do that

Reply:
Done at https://github.com/CQCL/pytket-quantinuum/issues/201
I have added that to pytket-quantinuum because this is an issues that needs to be solved for qasm, qir and on the hardware side